src/miniscript/context.rs

```diff
@@ -521,8 +529,15 @@ impl ScriptContext for Tap {
         // own unique signature. That is there is no way to re-use signatures for another
         // checksig. Therefore, for each successfully executed checksig, we will have
         // 64 bytes signature and thus sigops budget is always covered.
-        // There is overall limit of consensus
         // TODO: track height during execution
+        if let (Some(s), Some(h)) = (
+            ms.ext.exec_stack_elem_count_sat,
+            ms.ext.stack_elem_count_sat,
+        ) {
+            if s + h > MAX_STACK_SIZE {
```

Comment:
The check with initial elements + exec elements is done here.

Reply:
Should we fail if we have one of these properties and not the other?

Reply:
`None` here means that there is no satisfaction for this miniscript. In all cases, if one of them is Some, the other must also be Some

Reply:
Maybe should be Some((x,y)) instead of Some(x), Some(y) then? Or assert that `!a.xor(b).is_some()`?

Reply:
I will add the assert for now.